### PR TITLE
base64: add encode and decode to io, fixed #2964

### LINKED ISSUE #2964: Add streaming APIs for Base64
I'd like to be able to compile in gzipped assets as Base64 and then stream them directly to an IO via `Base64.decode(asset, io)`.  Right now I have to decode to a Slice and then stream the Slice which adds a copy.

### PR DESCRIPTION
**Pull request by kostya**

### CODE REVIEW

**Comment by asterite** [comment]
@kostya So fast!! 😍

File: src/base64.cr [hunk 1]
```diff
@@ -67,6 +67,28 @@ module Base64
     end
   end
 
+  # :nodoc:
+  def encode(data, io : IO)
+    inc = 0
```
Comment (asterite) [inline src/base64.cr]: @kostya This `inc`, `LINE_SIZE` and `NL` logic is duplicated, it also appears below. Could we extract it to a new method that tracks `inc` and yields the appropriate bytes? I don't know the name, probably the current `to_base64` should be `to_strict_base64` and this new one should be `to_base64`, but the names aren't very important. What do you think?

File: src/base64.cr [hunk 2]
```diff
@@ -67,6 +67,28 @@ module Base64
     end
   end
 
+  # :nodoc:
```
Comment (jhass) [inline src/base64.cr]: Why the `nodoc`s?

**Comment by asterite** [comment]
@kostya Thank you!!

**Pull request merged**